agent/proxycfg/state.go

```diff
@@ -443,7 +449,7 @@ func (s *state) initWatchesMeshGateway() error {
 			return err
 		}
 
-		err = s.cache.Notify(s.ctx, cachetype.HealthServicesName, &structs.ServiceSpecificRequest{
+		err = s.cache.Notify(s.ctx, s.serviceHealthCacheName, &structs.ServiceSpecificRequest{
```

Comment:
Note that this is just watching the consul service which is not Connect enabled - there is no `Connect: true` in the request below, so it should be using the non-Connect cache type with streaming.

Reply:
Never mind. I'd missed that we moved to a single cache type which just switches Topic based on the request: https://github.com/hashicorp/consul/blob/3760e3d12db55e9497a7fd3c8ef2e26c948439cc/agent/cache-types/streaming_health_services.go#L78-L80

That makes more sense!